CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(NanoLog VERSION 0.0.1 DESCRIPTION "Nanolog is an extremely performant nanosecond scale logging system for C++ that exposes a simple printf-like API.")
+
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/modules")
+
+set(CMAKE_CXX_STANDARD 17)
+
+# pre-requisites
+find_package(Threads REQUIRED)
+find_package(RT REQUIRED)
+
+file(GLOB HEADERS ${PROJECT_SOURCE_DIR}/runtime/*.h)
+file(GLOB SOURCES ${PROJECT_SOURCE_DIR}/runtime/*.cc ${PROJECT_SOURCE_DIR}/runtime/testHelper/GeneratedCode.cc)
+add_library(${PROJECT_NAME} ${SOURCES} ${HEADERS})
+
+target_include_directories(${PROJECT_NAME} PRIVATE ${PROJECT_SOURCE_DIR}/runtime)
```

Comment:
This should be marked as PUBLIC not PRIVATE to ensure that non standalone builds get the correct include directories.